core/configuration/dt_config_file_reader.go

```diff
@@ -58,7 +59,27 @@ type jsonConfigFileReader struct {
 // ReadConfigFromFile looks for a config file "dtconfig.json" in the current directory and attempts to parse it.
 // Returns an error if the file can't be read or the parsing fails.
 func (j *jsonConfigFileReader) readConfigFromFile() (fileConfig, error) {
-	return j.readConfigFromFileByPath("./dtconfig.json")
+	filePath := configFilePath()
+	return j.readConfigFromFileByPath(filePath)
+}
+
+func configFilePath() string {
+	if configFilePathFromEnv := os.Getenv("DT_CONFIG_FILE_PATH"); configFilePathFromEnv != "" {
+		return configFilePathFromEnv
+	}
+
+	// When running in a Google Cloud Functions Go runtime, we need to find the config file at a different path.
+	// For reference on the K_SERVICE environment variable, see:
+	// https://cloud.google.com/functions/docs/configuring/env-var#runtime_environment_variables_set_automatically
+	_, inGcf := os.LookupEnv("K_SERVICE")
```

Comment:
I'm fine with making this path conditional on presence of K_SERVICE, but IMHO we should still first look in the current directory and as fallback (if file cannot be opened) in this secondary directory.

Reply:
Why would we first check the wrong location and have 2 io operations instead of reading it from right location?

Reply:
In principle I'd agree with doing it like that, but I feel like in a cloud function setting, the more performant route may be preferrable. 

Though it may be a bit less flexible, if, for example, Google decides to change that internal folder structure. But since this folder structure is rather well-documented, I have doubts that they would introduce such a breaking change in the current generation of cloud functions.

Reply:
Can we rule out that there are deployments that have placed the file in the location where the pre-PR code would look for it and would no longer find it after we merge this?

Reply:
IIRC it's not possible to move the file anywhere but this `serverless_function_source_code` folder, which houses all source code files and resources such as our `dtconfig.json` file. Potentially the file could still be in a subfolder of that, but we weren't supporting that previously (we do now via the env variable).

Reply:
with fallback to `./dtconfig.json` all those deployments will still work as expected.

Reply:
Yes, I can add a fallback. I'm not 100% convinced that it's actually meaningful here since we've tested this case pretty well, but it's an additional bit of safety to ensure we don't introduce a breaking change of our own here.

Reply:
Added fallback logic in 76cc653e135d672f05002ccc67d2ab265e9a655b